### PR TITLE
feat: custom deploy addon support cluster addon and can select version

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:
feat custom deploy addon support cluster addon and can select version

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=635782&iterationID=12783&type=TASK)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     feat custom deploy addon support cluster addon and can select version         |
| 🇨🇳 中文    |       自定义部署Addon支持集群级别Addon并且支持版本选择       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: internal/tools/orchestrator/services/addon/addon.go [hunk 1]
```diff
@@ -2686,7 +2705,7 @@ func (a *Addon) ListCustomAddon() (*[]map[string]interface{}, error) {
 		addonMap["addonName"] = item.Name
 		addonMap["vars"] = addonSpec.ConfigVars
 		switch item.Name {
-		case "mysql", "api-gateway", "monitor":
+		case "mysql", "api-gateway", "monitor", apistructs.AddonRegisterCenter, apistructs.AddonNewConfigCenter:
```
Comment (iutx) [inline internal/tools/orchestrator/services/addon/addon.go:2708]: Is it possible to replace all of them with constants?

File: internal/tools/orchestrator/endpoints/addon.go [hunk 2]
```diff
@@ -71,6 +71,39 @@ func (e *Endpoints) CreateAddonDirectly(ctx context.Context, r *http.Request, va
...
+			hasConfig = true
+			if on.Options != nil && targetVersion == "" {
+				targetVersion = on.Options["version"]
+			}
+		}
+	}
+
+	if hasRegister && !hasConfig {
+		addonCreateReq.Addons["config-center"] = &diceyml.AddOn{
+			Plan: fmt.Sprintf("%s:%s", apistructs.AddonNewConfigCenter, "basic"),
```
Comment (iutx) [inline internal/tools/orchestrator/endpoints/addon.go:93]: Use constant `basic` from structs.

**Review by iutx — COMMENTED** [review]
@CeerDecy

**Review by iutx — APPROVED** [review]
LGTM @sfwn

File: apistructs/addon.go [hunk 3]
```diff
@@ -306,7 +306,7 @@ const (
 	// AddonConfigCenterAlias terminus-configcenter alias name
 	AddonConfigCenterAlias = "configcenter"
 	// AddonNewConfigCenter configcenter
-	AddonNewConfigCenter = "configcenter"
+	AddonNewConfigCenter = "config-center"
```
Comment (sfwn) [inline apistructs/addon.go:309]: why change `configcenter` to `config-center`?
but still `registercenter`, not `register-center`.

File: internal/tools/orchestrator/services/addon/addon.go [hunk 4]
```diff
@@ -2633,13 +2633,32 @@ func (a *Addon) ListCustomAddon() (*[]map[string]interface{}, error) {
...
+		apistructs.AddonMonitor,
+		apistructs.AddonRegisterCenter,
+		apistructs.AddonNewConfigCenter,
+	}
+	// if these addon-type is `custom` and they can't be deployed by user, skip it!
+	unCreateAbleAddons := []string{
+		apistructs.AddonMSENacos, // mse-nacos only can be deployed by `config-center` or `register-center`
+	}
+	createableAddonVersion := map[string]string{
+		apistructs.AddonApiGateway:      "3.0.0",
```
Comment (sfwn) [inline internal/tools/orchestrator/services/addon/addon.go:2648]: these default `addon:version` pairs should be putted in envs or db configs, for easier modifying.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/tools/orchestrator/services/addon/addon.go (hunk 4 above)
Comment (CeerDecy) [reply internal/tools/orchestrator/services/addon/addon.go]: oh yes, I found these version just for getting AddonSpec, now I've changed it to get AddonSpec from cache.

**Pull request merged**